test/y2partitioner/actions/resize_partition_test.rb

```diff
@@ -0,0 +1,186 @@
+
+    let(:min_size) { 100.KiB }
+
+    let(:max_size) { 1.GiB }
+
+    describe "#run" do
+      context "when the partition cannot be resized" do
+        let(:can_resize) { false }
+
+        it "shows an error popup" do
+          expect(Yast::Popup).to receive(:Error)
+          action.run
+        end
+
+        it "returns :back" do
+          expect(action.run).to eq(:back)
+        end
+      end
+
+      context "when the partition can be resized" do
+        let(:can_resize) { true }
+
+        context "and the user goes forward in the dialog" do
+          before do
+            allow(Y2Partitioner::Dialogs::PartitionResize).to receive(:run).and_return(:next)
+            partition.size = new_size
+          end
+
+          let(:new_size) { 1.GiB }
+
+          it "returns :finish" do
+            expect(action.run).to eq(:finish)
+          end
+
+          context "when the partition table does not require end-alignment" do
+            let(:scenario) { "mixed_disks" }
+
+            context "and the partition is end-aligned" do
+              let(:new_size) { 10.MiB }
+
+              it "does not change the partition size" do
+                size_before = partition.size
+
+                expect(partition.end_aligned?).to eq(true)
+                action.run
+                expect(partition.size).to eq(size_before)
+              end
+            end
+
+            context "and the partition is not end-aligned" do
+              let(:new_size) { 10.5.MiB }
+
+              it "aligns the partition" do
+                expect(partition.end_aligned?).to eq(false)
+                action.run
+                expect(partition.end_aligned?).to eq(true)
+              end
+            end
+          end
+
+          context "when the partition table requires end-alignment" do
+            let(:scenario) { "dasd_50GiB.yml" }
+
+            context "and the partition is end-aligned" do
+              let(:new_size) { 102432.KiB }
+
+              it "does not change the partition size" do
+                size_before = partition.size
+
+                expect(partition.end_aligned?).to eq(true)
+                action.run
+                expect(partition.size).to eq(size_before)
+              end
+            end
+
+            context "and the partition is not end-aligned" do
+              let(:new_size) { 12344.KiB }
+
+              it "aligns the partition" do
+                expect(partition.end_aligned?).to eq(false)
+                action.run
+                expect(partition.end_aligned?).to eq(true)
+              end
+            end
+          end
+        end
+
+        context "and the user aborts the process" do
+          before do
+            allow(Y2Partitioner::Dialogs::PartitionResize).to receive(:run).and_return(:abort)
+          end
+
+          it "returns :abort" do
+            expect(action.run).to eq(:abort)
+          end
+        end
+      end
+    end
+
+    describe "#fix_region_end" do
```

Comment:
A purist would tell you that you should test this using the public API instead of using `send` to call private methods. But I will not be that purist... today :wink: